### ISSUE TITLE
Add ability to set cocoapods version 

### ISSUE THREAD

**Issue by CHNB128** [react-native-community/react-native-circleci-orb#84]
## What would you like to be added

I would like to have ability to provide cocoapods version to use.

## Why is this needed

I have update my project to new Firebase SDK ^v4 and I got this error in ci run: 

```
[!] The version of CocoaPods used to generate the lockfile (1.10.1) is higher than the version of the current executable (1.8.4). Incompatibility issues may arise.
...
[!] `FirebaseAnalytics` requires CocoaPods version `>= 1.10.0`, which is not satisfied by your current version, `1.8.4`.
```

**Comment by matt-oakes** [comment]
The version of CocoaPods that is used is tied to the Xcode version in the CircleCI macOS executor that is used. It is [already possible to configure that](https://circleci.com/developer/orbs/orb/react-native-community/react-native#executors-macos) and the values you can use are documented here:

https://circleci.com/docs/2.0/testing-ios/#supported-xcode-versions

The default versions that were being used with this Orb were pretty out of date, so I've released v5.0.0 which updated Xcode to 12.4, CocoaPods to 1.10.1, and Node.js to 12. Either using the new defaults, or setting them yourself with the executor settings will solve your issue.

**Comment by saintpepsicola** [comment]
Hi @matt-oakes ,  

Problem is that I require the latest version of Cocoapod (1.11.0) which isn't _already_ installed in any of the Xcode versions. Circle CI recommends an alternate way of defining Cocoapod version in the iOS Podfile but that isn't picked up by this orb
